### PR TITLE
README changes for AppAuthTV

### PR DESCRIPTION
**Pull request by soule**
Still working on main README changes, included are the changes for the example project/Google IdP details.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@
 [![Pod Platform](https://img.shields.io/cocoapods/p/AppAuth.svg?style=flat)](https://cocoapods.org/pods/AppAuth)
 [![Catalyst compatible](https://img.shields.io/badge/Catalyst-compatible-brightgreen.svg?style=flat)](https://developer.apple.com/documentation/xcode/creating_a_mac_version_of_your_ipad_app)
 
-AppAuth for iOS and macOS is a client SDK for communicating with 
+AppAuth for iOS and macOS, and tvOS is a client SDK for communicating with
```
Comment (yalkhder) [inline README.md:11]:
```suggestion
AppAuth for iOS, macOS, and tvOS is a client SDK for communicating with 
```
Or we can simplify it to just "AppAuth for iOS", @WilliamDenniss thoughts?

File: README.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ custom URI scheme redirects are used. The library is friendly to other
 extensions (standard or otherwise), with the ability to handle additional params
 in all protocol requests and responses.
 
+For tvOS, AppAuth implements the [RFC 8628 - OAuth 2.0 Device Authorization Grant
```
Comment (yalkhder) [inline README.md:33]: nit: Just say OAuth 2.0 Authorization Grant.

```suggestion
For tvOS, AppAuth implements [OAuth 2.0 Device Authorization Grant
```

File: README.md [hunk 3]
```diff
@@ -44,20 +47,31 @@ Safari) on earlier versions.
 
 #### Authorization Server Requirements
 
-Both Custom URI Schemes (all supported versions of iOS) and Universal Links
-(iOS 9+) can be used with the library.
+AppAuthTV is designed for servers that support the device authorization flow as documented in
+native apps, as documented in [RFC 8628](https://tools.ietf.org/html/rfc8628).
```
Comment (yalkhder) [inline README.md:51]: Shouldn't this be under tvOS?

File: README.md [hunk 4]
```diff
@@ -82,7 +96,21 @@ client info to try the demo).
 
 ## Setup
 
-AppAuth supports four options for dependency management.
+AppAuth supports four options for dependency management. For **tvOS**, please use the 'TV' subspec in CocoaPods and `AppAuthTV` when using Swift Package Manager/Carthage.
```
Comment (yalkhder) [inline README.md:99]: I think the tvOS stuff should be under each respective package manager's section.

File: README.md (hunk 1 above)
Comment (WilliamDenniss) [reply README.md:11]: On the name, I think "AppAuth for iOS, macOS, and tvOS" is a bit of a mouthful, but does makes sense. Maybe this is the full name, and "AppAuth for iOS" is the short name?

File: README.md (hunk 4 above)
Comment (WilliamDenniss) [reply README.md:99]: +1, and it looks like it's covered in the version I see below

File: README.md [hunk 5]
```diff
@@ -144,6 +141,8 @@ Linked Framework and Libraries" section of your target).
 4. Add `AppAuth-iOS/Source` to your search paths of your target ("Build Settings ->
 "Header Search Paths").
 
+*Note: There is not a static library for AppAuthTV.*
```
Comment (yalkhder) [inline README.md:144]: nit: "There is no static library for AppAuthTV."

File: README.md [hunk 6]
```diff
@@ -47,12 +50,6 @@ Safari) on earlier versions.
 Both Custom URI Schemes (all supported versions of iOS) and Universal Links
 (iOS 9+) can be used with the library.
 
-In general, AppAuth can work with any authorization server that supports
-native apps, as documented in [RFC 8252](https://tools.ietf.org/html/rfc8252),
-either through custom URI scheme redirects, or universal links.
-Authorization servers that assume all clients are web-based, or require clients to maintain
-confidentiality of the client secrets may not work well.
-
 ### macOS
```
Comment (WilliamDenniss) [inline README.md:53]: Why is this removed?

Comment (soule) [reply README.md]: Sorry, accident.

File: README.md (hunk 1 above)
Comment (petea) [reply README.md:11]: There's currently (limited) watchOS support as well...  "AppAuth for Apple"?
